tests/e2e/monovertex.go

```diff
@@ -19,6 +20,27 @@ import (
 	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
 )
 
+func getMonoVertexName(namespace, monoVertexRolloutName string) string {
+
+	var monoVertexName string
+	label := fmt.Sprintf("%s,%s=%s", UpgradeStateLabelSelector, ParentRolloutLabel, monoVertexRolloutName)
+
+	Eventually(func() bool {
+		unstructList, err := dynamicClient.Resource(getGVRForMonoVertex()).Namespace(namespace).List(ctx, metav1.ListOptions{LabelSelector: label})
+		if err != nil {
+			return false
+		}
+		if len(unstructList.Items) == 0 {
+			return false
+		}
+		monoVertexName = unstructList.Items[0].GetName()
+		return true
+	}, 60*time.Second, testPollingInterval).Should(BeTrue())
```

Comment:
Maybe we don't need `Eventually` here anymore? Are all the calls to here within their own Eventually or Consistently block? Are all of them followed by a `Get()`? If so, could this just return the MonoVertex itself, and then the subsequent `Get()` doesn't need to be called?

Reply:
(same comment for Pipeline)

Reply:
Within `monovertex.go` we are either passing the name to another function or calling Get. In `functional_test.go`, we do need to get the Name to check if the MonoVertex is deleted or not.

Reply:
How about in the cases in which we are subsequently calling `Get()`, don't call `Get()` anymore and instead call `getMonoVertex()` which does not contain an Eventually call and just returns the `*Unstructured`? (and do let me know if I'm missing something)

Reply:
Got it! Should be addressed in my latest commit now.